editor/filesystem_dock.cpp

```diff
@@ -796,6 +796,19 @@ void FileSystemDock::_rename_operation(const String &p_to_path) {
 		EditorNode::get_singleton()->add_io_error("Error moving file:\n" + move_files[0] + "\n");
 	}
 
+	memdelete(da);
+	da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
```

Comment:
Why do you need filesystem access? `ACCESS_RESOURCES` as used above should be enough, and then you don't need to use magic numbers to strip `res://`.. (also you have a style issue with the extraneous space in the parentheses)
```gdscript
if (da->file_exists(move_files[0] + ".import") {
	err = da->rename(move_files[0] + ".import", p_to_path + ".import");	
	if (err != OK) {
		EditorNode::get_singleton()->add_io_error("Error moving file:\n" + move_files[0] + ".import" + "\n");
	}
}
```

Reply:
ACCESS_RESOURCES ignores the .import files since they aren't resources.

Reply:
Are you sure about that? I don't see any logic in DirAccess to define whether files are resources or not, `ACCESS_RESOURCES` just seems to mean access the `res://` folder.

Reply:
That's what I thought too, but simply appending ".import" causes it to not see the import files at all.

```
if (da->file_exists(move_files[0] + ".import") {
```
will return false every time even though something like "res://icon.png.import" exists.

Reply:
I'm going to try and hardcode in a path that I know exists and see if it sees the file, just to be sure.

Reply:
Hmmm, I must've been doing something wrong before when I was using the ACCESS_RESOURCES mode because it's working now. Thanks Rémi :)